### DISCUSSION TITLE
HomeAssisant Intergration

### DISCUSSION THREAD

**Discussion by Next9999** [KrystianD/bm2-battery-monitor#2] [General]
Hello

Do you plan or think that it could be integrated with Home Assistant?

**Comment by KrystianD** [comment]
Hi @Next9999,

Integration with HA can be done using MQTT or by configuring esphome project [for use with HA](https://esphome.io/guides/getting_started_hassio.html).

**Reply by Next9999** [reply]
Hello

i am new to HA, look like i need to purchase a ESP32 for that

Andre

**Comment by Next9999** [comment]
ESp8266 install  working with a simple thermometer dht11

Any hint on how should the BM2 be integrated in Home Assistant

Thks

**Comment by Next9999** [comment]
i have copy/modify the yaml as bellow, when i run it the log only give this 

Any help would apreaciated

INFO Reading configuration /config/esphome/bm2.yaml...
INFO Starting log output from esphome/bm2/debug
INFO Connected to MQTT broker!
INFO Successfully reconnected to the MQTT server
INFO Connected to MQTT broker!
INFO Successfully reconnected to the MQTT server



====================================
esphome:
  name: bm2
  
#  includes:
#    - include.h
#include "mbedtls/aes.h"

esp32:
  board: nodemcu-32s
  framework:
    type: arduino

logger:
  level: INFO

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

mqtt:
  broker: !secret mqtt_hostname
  discovery: false
  topic_prefix: esphome/bm2

web_server:
  port: 80

esp32_ble_tracker:

ble_client:
  - mac_address: !secret bm2_mac
    id: bm2_battery_meter

sensor:
  - platform: ble_client
    ble_client_id: bm2_battery_meter
    name: Car_Voltage
    service_uuid: 'fff0'
    characteristic_uuid: 'fff4'
    unit_of_measurement: 'V'
    accuracy_decimals: 2
    state_class: measurement
    device_class: voltage
    force_update: true
    expire_after: 5min
    notify: true
    lambda: |-
      mbedtls_aes_context aes;
      mbedtls_aes_init(&aes);
      unsigned char output[16];
      unsigned char key[16] = { 108, 101, 97, 103, 101, 110, 100, 255, 254, 49, 56, 56, 50, 52, 54, 54, };
      unsigned char iv[16] = {};
      mbedtls_aes_setkey_dec(&aes, key, 128);
      mbedtls_aes_crypt_cbc(&aes, MBEDTLS_AES_DECRYPT, 16, iv, (uint8_t*)&x[0], output);
      mbedtls_aes_free(&aes);
      return ((output[2] | (output[1] << 8)) >> 4) / 100.0f;

**Comment by adrianmihalko** [comment]
Here is how I integrated without mqtt with ESP32: https://community.home-assistant.io/t/bm2-battery-monitoring-using-ble-tracker-component/408865/11?u=madrian

Thanks for the great solution.

**Reply by Next9999** [reply]
i spend a lot of time... on this :-(   Any help would be appreciated... i really like to add BM2 to my Motorhome dashboard

![image](https://user-images.githubusercontent.com/54117519/190880685-c57e3123-e8ac-48b8-9eef-e3c32b46c850.png)
